pyproject.toml

```diff
@@ -64,7 +62,18 @@ Sphinx = ">=5,<8"
 sphinx-rtd-theme = ">=1,<2"
 sphinx-tabs = ">=3.4.0,<4"
 sphinx-toolbox = "^3"
-isort = "5.10.1"
+
+[tool.ruff]
+line-length = 120
+# Enable pycodestyle (`E`) and Pyflakes (`F`) codes.
+lint.select = ["E", "F"]
+# Ignore E501 line-too-long - see https://docs.astral.sh/ruff/faq/#is-the-ruff-linter-compatible-with-black for why
+lint.ignore = ["F405", "E501", "E203", "E731", "N818"]
```

Comment:
Why ignore these extras (other than E731 which is a pain and I agree should be ignored!)?

Note that it's possible to do more finely grained ignores than this for, for example, F405. An example of ignoring in django settings files (which could easily be recrafted if all you wanted to do was ignore particular lint warning sin test files, for example
```

[tool.ruff.lint.per-file-ignores]
# Allow variables from use of `import *` in settings files in django
"backend/settings*" = ["F405"]
```

Reply:
I copied them over from `setup.cfg`. I can't remember the exact details but they were messing up the code readability in weird ways a while ago so I disabled them. It was stuff to do with binary operators I think